fhir/client.go

```diff
@@ -20,23 +20,39 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"path"
 )
 
 // A Client is a FHIR client which combines an HTTP client with the base URL of
-// a FHIR server. At minimum, the Base has to be set. HttpClient can be left at
+// a FHIR server. At minimum, the BaseURL has to be set. HttpClient can be left at
 // its default value.
 type Client struct {
-	HttpClient        http.Client
-	Base              string
+	httpClient http.Client
+	baseURL    *url.URL
```

Comment:
In order to get the relative path resolving to work, the baseURL has to end with a slash.

Reply:
Can confirm, thanks for the hint. Additional code ensuring the existing base URL's path will end with a slash on client creation.